tests/TestCase/ORM/TableTest.php

```diff
@@ -5808,6 +5808,26 @@ public function testFindOrCreatePartialValidation(): void
         $articles->findOrCreate(['body' => 'test']);
     }
 
+    /**
+     * Test that findOrCreate with array data.
+     */
+    public function testFindOrCreateArrayData(): void
+    {
+        $articles = $this->getTableLocator()->get('Articles');
+
+        $firstArticle = $articles->findOrCreate(['title' => 'Not there'], ['body' => 'New body']);
+        $this->assertFalse($firstArticle->isNew());
+        $this->assertNotNull($firstArticle->id);
+        $this->assertSame('Not there', $firstArticle->title);
+        $this->assertSame('New body', $firstArticle->body);
+
+        $secondArticle = $articles->findOrCreate(['title' => 'Not there'], ['body' => 'New body']);
+        $this->assertFalse($secondArticle->isNew());
+        $this->assertNotNull($secondArticle->id);
+        $this->assertSame('Not there', $secondArticle->title);
```

Comment:
Should this scenario have a different `body` and assert that the body has not changed?